### PR TITLE
[CI] Centos build in ci

### PR DESCRIPTION
**Pull request by lachmatt**
## Why

Towards #2083 

## What

- include centos in CI container build
- rename dockerfiles

## Tests
CI

## Checklist

~~- [ ] `CHANGELOG.md` is updated.~~
~~- [ ] Documentation is updated.~~
~~- [ ] New features are covered by tests.~~

### CODE REVIEW

**Review by Kielek — APPROVED** [review]

**Comment by Kielek** [comment]
@open-telemetry/dotnet-instrumentation-maintainers, some adjustment in required builds are needed we will be ok to merge it.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -93,22 +93,26 @@ jobs:
...
+        docker build -t mybuildimage -f "./docker/${{ matrix.base-image }}.dockerfile" ./docker
+        docker run -e OS_TYPE=${{ matrix.os-type }} --rm --mount type=bind,source="${GITHUB_WORKSPACE}",target=/project mybuildimage \
           ./build.sh Workflow --containers none
     - name: Test the Shell scripts from README.md in Docker container
       run: |
         set -e
-        docker build -t mybuildimage -f "./docker/${{ matrix.base-image }}.dockerfile" .
-        docker run -e OS_TYPE=linux-musl --mount type=bind,source="${GITHUB_WORKSPACE}",target=/project --rm mybuildimage /bin/sh -c '
+        docker build -t mybuildimage -f "./docker/${{ matrix.base-image }}.dockerfile" ./docker
+        docker run -e OS_TYPE=${{ matrix.os-type }} --mount type=bind,source="${GITHUB_WORKSPACE}",target=/project --rm mybuildimage /bin/sh -c '
```
Comment (pellared) [inline .github/workflows/ci.yml:115]:
```suggestion
        docker run -e --mount type=bind,source="${GITHUB_WORKSPACE}",target=/project --rm mybuildimage /bin/sh -c '
```

Comment (pellared) [reply .github/workflows/ci.yml:115]: OS_TYPE should be not needed here

**Review by pellared — APPROVED** [review]

File: .github/workflows/ci.yml (hunk 1 above)
Comment (lachmatt) [reply .github/workflows/ci.yml]: removed in https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/pull/2193/commits/8ec9efa9cc818523f030d9c2606bb614f6650d71

**Comment by Kielek** [comment]
image
marked as required.
Previous job removed from the same list.

**Pull request merged**